pkg/controller/perconaservermongodb/pbm.go

```diff
@@ -0,0 +1,503 @@
+	if err != nil {
+		if errors.Is(err, bsoncore.ErrElementNotFound) {
+			val = false
+		} else {
+			return errors.Wrap(err, "get pitr.oplogOnly")
+		}
+	}
+
+	oplogOnly, ok := val.(bool)
+	if !ok {
+		return errors.Errorf("unexpected value of pitr.oplogOnly: %T", val)
+	}
+
+	if oplogOnly == cr.Spec.Backup.PITR.OplogOnly {
+		return nil
+	}
+
+	enabled := strconv.FormatBool(cr.Spec.Backup.PITR.OplogOnly)
+
+	log.Info("Setting pitr.oplogOnly", "value", enabled)
+
+	if err := pbm.SetConfigVar(ctx, "pitr.oplogOnly", enabled); err != nil {
+		return errors.Wrap(err, "update pitr.oplogOnly")
+	}
+
+	return nil
+}
+
+func updateOplogSpanMinIfNeeded(ctx context.Context, pbm backup.PBM, cr *psmdbv1.PerconaServerMongoDB) error {
+	log := logf.FromContext(ctx)
+
+	val, err := pbm.GetConfigVar(ctx, "pitr.oplogSpanMin")
+	if err != nil {
+		if errors.Is(err, mongo.ErrNoDocuments) {
+			return nil
+		}
+		// PBM-1387
+		// PBM has a bug to return config fields if they use default value
+		if errors.Is(err, bsoncore.ErrElementNotFound) {
+			val = 0.0
+		} else {
+			return errors.Wrap(err, "get pitr.oplogSpanMin")
+		}
+	}
+
+	oplogSpanMin, ok := val.(float64)
+	if !ok {
+		return errors.Errorf("unexpected value of pitr.oplogSpanMin: %T", val)
+	}
+
+	if oplogSpanMin == cr.Spec.Backup.PITR.OplogSpanMin.Float64() {
+		return nil
+	}
+
+	valStr := cr.Spec.Backup.PITR.OplogSpanMin.String()
+
+	log.Info("Setting pitr.oplogSpanMin", "value", valStr)
+
+	if err := pbm.SetConfigVar(ctx, "pitr.oplogSpanMin", valStr); err != nil {
+		return errors.Wrap(err, "update pitr.oplogSpanMin")
+	}
+
+	return nil
+}
+
+func updateCompressionIfNeeded(ctx context.Context, pbm backup.PBM, cr *psmdbv1.PerconaServerMongoDB) error {
+	log := logf.FromContext(ctx)
+
+	val, err := pbm.GetConfigVar(ctx, "pitr.compression")
+	compression := ""
+	if err != nil {
+		if errors.Is(err, mongo.ErrNoDocuments) {
+			return nil
+		} else if !errors.Is(err, bsoncore.ErrElementNotFound) {
+			return errors.Wrap(err, "get pitr.compression")
+		}
+	} else {
+		var ok bool
+		compression, ok = val.(string)
+		if !ok {
+			return errors.Errorf("unexpected value of pitr.compression: %T", val)
+		}
+	}
+
+	if compression != string(cr.Spec.Backup.PITR.CompressionType) {
+		if string(cr.Spec.Backup.PITR.CompressionType) == "" {
+			cfg, err := pbm.GetConfig(ctx)
+			if err != nil {
+				return errors.Wrap(err, "get pbm config")
+			}
+
+			cfg.PITR.Compression = ""
+			log.Info("Delete pitr.compression")
+			if err := pbm.SetConfig(ctx, cfg); err != nil {
+				return errors.Wrap(err, "delete pitr.compression")
+			}
+		} else {
+			val := string(cr.Spec.Backup.PITR.CompressionType)
+			err := pbm.SetConfigVar(ctx, "pitr.compression", val)
+			log.Info("Setting pitr.compression", "value", val)
+			if err != nil {
+				return errors.Wrap(err, "update pitr.compression")
+			}
+		}
+
+		// PBM needs to disabling and enabling PITR to change compression type
```

Comment:
typo: `		// PBM needs to disable and enabling PITR to change compression type
`